crates/matrix-sdk-crypto/src/session_manager/group_sessions.rs

```diff
@@ -379,8 +379,18 @@ impl GroupSessionManager {
         // This is calculated in the following code and stored in this variable.
         let mut should_rotate = user_left || visibility_changed || algorithm_changed;
 
+        let own_identity =
+            self.store.get_user_identity(self.store.user_id()).await?.and_then(|i| i.into_own());
+
         for user_id in users {
-            let user_devices = self.store.get_user_devices_filtered(user_id).await?;
+            let user_devices = self.store.get_readonly_devices_filtered(user_id).await?;
+
+            // We only need the user identity if settings.only_allow_trusted_devices is set.
```

Comment:
Presumably this is the part that improved the performance of things? I wonder if a simple cache, so we only load the `device_owner_identity` once would have given us the same performance improvement as well.

Presumably no, since the memory store shows a 5% improvement as well. Any thoughts on this?

Reply:
Yes, this is the performance improvement. It's partly not requesting the owning user's identity, and partly not requesting our own identity on each iteration of the loop.